test/load_transactions_use_case_tests.dart

```diff
@@ -0,0 +1,121 @@
+import 'dart:async';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:simple_result/simple_result.dart';
+import 'package:transactions/Feature/LocalTransactionLoader.dart';
+import 'package:transactions/Feature/Transaction.dart';
+import 'package:transactions/Store/TransactionStore.dart';
+import 'package:tuple/tuple.dart';
+
+void main() {
+  test("test_init_doesNotLoadAnyTransactions", () {
+    final tuple = _makeSUT();
+    final store = tuple.item2;
+
+    expect(store.messages.isEmpty, true);
+  });
+
+  test("test_load_requestLoadTransactions", () {
+    final tuple = _makeSUT();
+    final sut = tuple.item1;
+    final store = tuple.item2;
+
+    sut.load();
+
+    expect(store.messages, [TransactionStoreSpyMessage.load]);
+  });
+
+  test("test_loadTwice_requestLoadTransactionsTwice", () {
+    final tuple = _makeSUT();
+    final sut = tuple.item1;
+    final store = tuple.item2;
+
+    sut.load();
+    sut.load();
+
+    expect(store.messages, [TransactionStoreSpyMessage.load, TransactionStoreSpyMessage.load]);
+  });
+
+  test("test_load_deliversErrorOnLoadTransactionsError", () async {
+    final expectedError = TransactionStoreUnknownException();
+    final store = TransactionStoreStub(Result.failure(expectedError));
+    final sut = LocalTransactionLoader(store);
+
+    final result = sut.load();
+    result
+        .then((value) => () {
+              assert(false, "Expect failure, got : " + value.toString());
+            })
+        .catchError((error) {
+      expect(error, TransactionStoreUnknownException);
+    });
+  });
+
+  test("test_load_succeedsWithEmptyTransactions", () {
+    final store = TransactionStoreStub(Result.success([]));
+    final sut = LocalTransactionLoader(store);
+
+    final result = sut.load();
+    result
+        .then((value) => () {
+              value.when(success: (transactions) {
+                expect(transactions.isEmpty, true);
+              }, failure: (error) {
+                assert(false, "Expect to complete with success, got " + error.toString() + "instead.");
+              });
+            })
+        .catchError((error) {
+      assert(false, "Expect to complete with success, got " + error + "instead.");
+    });
+  });
+
+  test("test_load_succeedsWithTransactions", () {
```

Comment:
This test doesn't have expect(). Is this written as intended?

Reply:
Oops, I think it is better to use expect. I will change it.

Reply:
Oh I just notice that there is expect method on the success path.

Reply:
So it is actually valid